internal/pkg/vault/managment_test.go

```diff
@@ -0,0 +1,407 @@
+	require.NoError(t, err)
+}
+
+func TestInstallPolicy(t *testing.T) {
+	mockLogger := logger.MockLogger{}
+	expected := "policydoc"
+
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, http.MethodPut, r.Method)
+		require.Equal(t, "/v1/sys/policies/acl/policy-name", r.URL.EscapedPath())
+		require.Equal(t, expectedToken, r.Header.Get(AuthTypeHeader))
+
+		// Make sure the policy doc was base64 encoded in the json response object
+		body := make(map[string]interface{})
+		err := json.NewDecoder(r.Body).Decode(&body)
+		require.NoError(t, err)
+		require.Equal(t, expected, body["policy"].(string))
+
+		w.WriteHeader(http.StatusNoContent)
+	}))
+	defer ts.Close()
+
+	client := createClient(t, ts.URL, mockLogger)
+
+	// Act
+	err := client.InstallPolicy(expectedToken, "policy-name", expected)
+
+	// Assert
+	require.NoError(t, err)
+}
+
+func TestCheckSecretEngineInstalled(t *testing.T) {
+	// Arrange
+	mockLogger := logger.MockLogger{}
+
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, http.MethodGet, r.Method)
+		require.Equal(t, MountsAPI, r.URL.EscapedPath())
+		require.Equal(t, expectedToken, r.Header.Get(AuthTypeHeader))
+
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte(`{
+			"data": {
+				"cubbyhole/": {
+					"accessor": "cubbyhole_23676773",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "per-token private secret storage",
+					"local": true,
+					"options": null,
+					"seal_wrap": false,
+					"type": "cubbyhole"
+				},
+				"identity/": {
+					"accessor": "identity_11e23ad0",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "identity store",
+					"local": false,
+					"options": null,
+					"seal_wrap": false,
+					"type": "identity"
+				},
+				"secret/": {
+					"accessor": "kv_3ee7b0c0",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "key/value secret storage",
+					"local": false,
+					"options": {
+						"version": "1"
+					},
+					"seal_wrap": false,
+					"type": "kv"
+				},
+				"sys/": {
+					"accessor": "system_5e0c411d",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "system endpoints used for control, policy and debugging",
+					"local": false,
+					"options": null,
+					"seal_wrap": false,
+					"type": "system"
+				}
+			}	
+		  }`))
+		require.NoError(t, err)
+
+	}))
+	defer ts.Close()
+
+	client := createClient(t, ts.URL, mockLogger)
+
+	// Act
+	installed, err := client.CheckSecretEngineInstalled(expectedToken, "secret/", "kv")
+
+	// Assert
+	require.NoError(t, err)
+	assert.True(t, installed)
```

Comment:
I've added more tests for check engine installed on consul as well, please modify this test based on that.
https://github.com/edgexfoundry/edgex-go/pull/3179/files#diff-87dbdd4059b58a71c40514e7ed80b05bf39658cdc7548d498004faad697397aa

Reply:
`managmetn_test` --> `management_test`